scripts/BumpVersion.swift

```diff
@@ -0,0 +1,169 @@
+import Foundation
+
+let greenColor = "\u{001B}[0;32m"
+let resetColor = "\u{001B}[0;0m"
+let redColor = "\u{001B}[0;31m"
+let yellowColor = "\u{001B}[33m"
+
+guard CommandLine.arguments.count == 2 else {
+    print("\(redColor)Please provide a version in the format `major.minor.patch`\(resetColor)")
+    exit(1)
+}
+
+func isVersionBumped(_ currentVersion: String, _ newVersion: String) -> Bool {
+    let currentComponents = currentVersion.components(separatedBy: ".")
+    let newComponents = newVersion.components(separatedBy: ".")
+
+    guard currentComponents.count == 3 && newComponents.count == 3 else {
+        return false
+    }
+
+    let currentMajor = Int(currentComponents[0]) ?? 0
+    let currentMinor = Int(currentComponents[1]) ?? 0
+    let currentPatch = Int(currentComponents[2]) ?? 0
+
+    let newMajor = Int(newComponents[0]) ?? 0
+    let newMinor = Int(newComponents[1]) ?? 0
+    let newPatch = Int(newComponents[2]) ?? 0
+
+    if newMajor < currentMajor {
+        return false
+    }
+
+    if newMajor == currentMajor && newMinor < currentMinor {
+        return false
+    }
+
+    if newMajor - currentMajor > 1 {
+        print("\(yellowColor)Warning: New version major number should not jump by more than 1.\(redColor)")
+    }
+
+    if newMajor > currentMajor  {
+        if newMinor != 0 {
+            print("\(yellowColor)Warning: New major version should have a minor number of 0.\(resetColor)")
+        } else if newPatch != 0 {
+            print("\(yellowColor)Warning: New major version should have a patch number of 0.\(resetColor)")
+        }
+    }
+
+    if newMinor - currentMinor > 1 {
+        print("\(yellowColor)Warning: New version minor number should not jump by more than 1.\(resetColor)")
+    } else if newMinor > currentMinor && newPatch != 0 {
+        print("\(yellowColor)Warning: New minor version should have a patch number of 0.\(resetColor)")
+    }
+
+    if newPatch - currentPatch > 1 {
+        print("\(yellowColor)Warning: New version patch number should not jump by more than 1.\(resetColor)")
+    }
```

Comment:
I wonder if it might be better to just take in `-M, --major`, `-m, --minor`, and `-p, --patch`, and just bump the correct number instead of doing all the work to parse input and validate it.

Reply:
That's a good point, I think it would've made the implementation easier from the beginning for sure. But since input validation and parsing is already implemented, I think just taking in the version as a single argument is the easier option for users.